sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/analysis/TypeCoercion.scala

```diff
@@ -994,8 +994,12 @@ object TypeCoercion extends TypeCoercionBase {
       case (StringType, datetime: DatetimeType) => datetime
       case (StringType, AnyTimestampType) => AnyTimestampType.defaultConcreteType
       case (StringType, BinaryType) => BinaryType
+      case (st: StringType, StringType) => st
+      case (st: StringType, _: StringTypeCollated) => st
```

Comment:
I think this will be covered by the last default case match?

Reply:
Yes and no. The following two lines would have made a cast, but I changed them so they doesn't.